File: app/src/main/java/com/boost/espressotest/presentation/screen/detail/view/DetailView.java
package com.boost.espressotest.presentation.screen.detail.view;

import com.boost.espressotest.domain.model.Product;
import com.boost.espressotest.presentation.base.BaseView;

/**
 * @author PerSpiKyliaTor on 19.01.18.
 */

public interface DetailView extends BaseView {
    void showProduct(Product product);

    void productLoadError();

    void updateProductStatus(boolean isFavorite);

    void productStatusUpdateError();
}

File: app/src/main/java/com/boost/espressotest/presentation/screen/detail/view/DetailView.kt
package com.boost.espressotest.presentation.screen.detail.view

import com.boost.espressotest.domain.model.Product
import com.boost.espressotest.presentation.base.BaseView

interface DetailView : BaseView {
    fun showProduct(product: Product)

    fun productLoadError(error: String? = null)

    fun updateProductStatus(isFavorite: Boolean)

    fun productStatusUpdateError(error: String? = null)
}
